sway/tree/view.c

```diff
@@ -770,3 +771,80 @@ bool view_has_mark(struct sway_view *view, char *mark) {
 	}
 	return false;
 }
+
+static void update_marks_texture(struct sway_view *view,
+		struct wlr_texture **texture, struct border_colors *class) {
+	struct sway_container *output = container_parent(view->swayc, C_OUTPUT);
+	if (!output) {
+		return;
+	}
+	if (*texture) {
+		wlr_texture_destroy(*texture);
+	}
+	if (!view->marks->length) {
```

Comment:
unmark results in a segmentation fault. Changing this to `!view->marks || !view->marks->length` fixes the issue.

Reply:
Thanks. I think I've fixed this issue, but not using the suggestion you gave. `view->marks` should always be a valid list.

I replicated a segfault by marking a view then running `swaymsg unmark`. I realised I need to set `*texture` to `NULL` after destroying it. After doing that, unmark works correctly. I also copied this fix into the title texture code too - I guess it's not very common to update a title to be `NULL` which is why we haven't discovered it already.